jolt-core/src/r1cs/spartan.rs

```diff
@@ -305,16 +311,29 @@ impl<F: PrimeField, G: CurveGroup<ScalarField = F>> UniformSpartanProof<F, G> {
                         + small_C_evals[i] * r_sq
                 })
                 .collect::<Vec<F>>();
+            drop(_enter);
 
             // 2. Handles all entries but the last one with the constant 1 variable
-            let mut RLC_evals: Vec<F> = (0..key.num_vars_total)
-                .into_par_iter()
-                .map(|col| eq_rx_ts[col % n_steps] * small_RLC_evals[col / n_steps])
-                .collect();
-            let next_pow_2 = 2 * key.num_vars_total;
-            RLC_evals.resize(next_pow_2, F::zero());
+            let other_span = tracing::span!(tracing::Level::TRACE, "poly_ABC_wait_alloc_complete");
+            let _other_enter = other_span.enter();
+            let mut RLC_evals = RLC_evals_alloc.join().unwrap();
+            drop(_other_enter);
+
+            let span = tracing::span!(tracing::Level::TRACE, "poly_ABC_big_RLC_evals");
+            let _enter = span.enter();
+            // small_RLC_evals is 30% ones.
+            RLC_evals.par_chunks_mut(n_steps).take(key.num_vars_total / n_steps).enumerate().for_each(|(chunk_index, rlc_chunk)| {
+                if !small_RLC_evals[chunk_index].is_zero() {
+                    for (eq_index, item) in rlc_chunk.iter_mut().enumerate() {
+                        *item = eq_rx_ts[eq_index] * small_RLC_evals[chunk_index];
+                    }
+                }
+            });
```

Comment:
Did this help? I'm pretty I tried something similar but it was slower

Reply:
high variance but 25% to 50% quicker. small_RLC_evals is also 25% zeros